internal/server/proto/server.proto

```diff
@@ -2144,16 +2146,34 @@ message Runner {
   // jobs from being assigned to them.
   bool by_id_only = 2;
 
-  // odr is true if this runner is an on-demand runner. For ODRs, we expect
-  // they will accept exactly one job and then exit. This is used by the
-  // server to change some other behavior:
-  //
-  //   * The server will give ODRs project/app-scoped config if it exists.
-  //   * The server will never assign more than one job to this runner.
-  //     This is also enforced in the runner client-side but the server also
-  //     does this out of caution.
-  //
-  bool odr = 4;
+  // Kind indicates what variety of runner this is
+  oneof kind {
+
+    // odr is set if this runner as an on-demand runner. For ODRs, we expect
+    // they will accept exactly one job and then exit. This is used by the
+    // server to change some other behavior:
+    //
+    //   * The server will give ODRs project/app-scoped config if it exists.
+    //   * The server will never assign more than one job to this runner.
+    //     This is also enforced in the runner client-side but the server also
+    //     does this out of caution.
+    //
+    ODR odr = 5;
+
+    // local indicates this runner was created by a cli instantiation
+    Local local = 6;
```

Comment:
Instead of empty messages you can use `google.protobuf.Empty` here if you want. You can change to a message type later and be fully backwards compatible on the wire. Up to you.

Reply:
If they were intended solely as markers i'd switch, but I imagine we'll want to add more fields to them in the not-too-distant future.